titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java

```diff
@@ -219,20 +199,124 @@ public void enterActiveMode() {
+                        logger.debug("Node Deleted: {}, deletedFinalStateUnknown={}", node, deletedFinalStateUnknown);
+                        String leaseId = node.getMetadata().getName();
+                        logger.info("Removing lease: {}", leaseId);
+                        rescindLeaseHandler.call(Collections.singletonList(LeaseRescindedEvent.leaseIdEvent(leaseId)));
+                    }
+                });
+
+        return nodeInformer;
+    }
+
+    private SharedIndexInformer<V1Pod> createPodInformer(SharedInformerFactory sharedInformerFactory, CoreV1Api api) {
+        SharedIndexInformer<V1Pod> podInformer =
+                sharedInformerFactory.sharedIndexInformerFor(
+                        (CallGeneratorParams params) -> api.listNamespacedPodCall(
+                                KUBERNETES_NAMESPACE,
+                                null,
+                                null,
+                                null,
+                                null,
+                                null,
+                                params.resourceVersion,
+                                params.timeoutSeconds,
+                                params.watch,
+                                null,
+                                null
+                        ),
+                        V1Pod.class,
+                        V1PodList.class,
+                        RESYNC_PERIOD_MS
+                );
+
+        podInformer.addEventHandler(
+                new ResourceEventHandler<V1Pod>() {
+                    @Override
+                    public void onAdd(V1Pod pod) {
+                        podUpdated(pod);
+                    }
+
+                    @Override
+                    public void onUpdate(V1Pod oldPod, V1Pod newPod) {
+                        podUpdated(newPod);
+                    }
+
+                    @Override
+                    public void onDelete(V1Pod pod, boolean deletedFinalStateUnknown) {
+                        podUpdated(pod);
+                    }
+                });
+
+        return podInformer;
+    }
+
+    @Deactivator
+    public void shutdown() {
+        sharedInformerFactory.stopAllRegisteredInformers();
+    }
+
     @Override
     public void launchTasks(List<Protos.TaskInfo> requests, List<VirtualMachineLease> leases) {
         for (Protos.TaskInfo request : requests) {
-            long startTimeMs = clock.wallTime();
             try {
                 V1Pod v1Pod = taskInfoToPod(request);
                 logger.info("creating pod: {}", v1Pod);
-                normalApi.createNamespacedPod(KUBERNETES_NAMESPACE, v1Pod, null, null, null);
-                podsClientMetrics.incrementOnSuccess(POST, PODS, STATUS_200);
-                podsClientMetrics.registerOnSuccessLatency(POST, Duration.ofMillis(clock.wallTime() - startTimeMs));
+                api.createNamespacedPod(KUBERNETES_NAMESPACE, v1Pod, null, null, null);
```

Comment:
Are these metrics replaced with something else?

Reply:
yes, I instrumented the `OkHttp` client so all the calls has metrics.